src/either.rs

```diff
@@ -0,0 +1,82 @@
+//! Extension of [`either::Either`] to provide support for [`NonEmptyIterator`].
+//!
+//! ```
+//! use nonempty_collections::*;
+//! fn get_data(input: usize) -> NEEither<[usize; 1], [usize; 3]> {
+//!     if input == 0 {
+//!         NEEither::Left([0])
+//!     } else {
+//!         NEEither::Right([2, 1, 4])
+//!     }
+//! }
+//!
+//! assert_eq!(
+//!     nev![0],
+//!     get_data(0).into_nonempty_iter().collect::<NEVec<_>>()
+//! );
+//! ```
+use either::Either;
+
+use crate::IntoNonEmptyIterator;
+use crate::NonEmptyIterator;
+
+/// Non-empty variant of [`either::Either`] that implements
+/// [`NonEmptyIterator`].
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub enum NEEither<L, R> {
+    /// A value of type `L`.
+    Left(L),
+    /// A value of type `R`.
+    Right(R),
+}
```

Comment:
Does this type need to exist? Why can't we just implement `NonEmptyIterator` directly on `Either`?